src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -1467,24 +1470,24 @@ App::DocumentObjectExecReturn *Hole::execute(void)
             const double pi = 3.14159265359;
             bool leftHanded = (bool) ThreadDirection.getValue();
 
-            // Nomenclature and formulae according to Schaeffler Technical pocket guide (edition 2018)
-            // this is the same for all metric and UTS threads.
-            double D = threadDescription[ThreadType.getValue()][ThreadSize.getValue()].diameter;
-            double P = ThreadPitch.getValue();
-            double H1 = 0.54127 * P;
-            double H = 0.86603 * P;
-            double D1 = D - 2 * H1;
-            double alfa = 60.0 * pi/180;
-
-
-            double clearance;   // clearance used for Actual Thread modeling.
+            // Nomenclature and formulae according to Figure 1 of ISO 68-1
+            // this is the same for all metric and UTS threads as stated here:
+            // https://en.wikipedia.org/wiki/File:ISO_and_UTS_Thread_Dimensions.svg
+            // Note that in the ISO standard, Dmaj is called D, which has been followed here.
+            double D = threadDescription[ThreadType.getValue()][ThreadSize.getValue()].diameter;  //
+            double P = ThreadPitch.getValue();            // Pitch
+            double H = sqrt(3.0)/2.0 * P;                 // Height of fundamentral triangle
+            double H1 = 5*H/8 * P;                        // The thread depth
+            double D1 = D - 2 * H1;                       // Basic minor diameter of internal thread
+            double alfa = Base::toRadians(60.0);          // Angle of the fundamental triangle. Same for both ISO and UTS.

```

Comment:
Nitpicking ;-): why not "alpha" instead of "alfa"?

Reply:
Old habit. Anyway, this is gone now.